Watchman.Discord/Areas/Administration/Controllers/MutedUsersController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Devscord.DiscordFramework.Framework.Architecture.Controllers;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using Devscord.DiscordFramework.Services.Factories;
+using Watchman.Discord.Areas.Administration.BotCommands;
+using Watchman.Discord.Areas.Protection.Services;
+using Watchman.DomainModel.Messages;
+
+namespace Watchman.Discord.Areas.Administration.Controllers
+{
+    public class MutedUsersController : IController
+    {
+        private readonly MutingHelper _mutingHelper;
+        private readonly UsersService _usersService;
+        private readonly DirectMessagesService _directMessagesService;
+
+        public MutedUsersController(DirectMessagesService directMessagesService, MutingHelper mutingHelper, UsersService usersService)
+        {
+            this._directMessagesService = directMessagesService;
+            this._mutingHelper = mutingHelper;
+            this._usersService = usersService;
+        }
+
+        [AdminCommand]
+        public async Task MutedUsers(MutedUsersCommand mutedUsersCommand, Contexts contexts)
+        {
+            var mutedUsersList = await _usersService.GetUsersAsync(contexts.Server).ToListAsync();
+            var (title, description, values) = this.GetMuteEmbedMessage(mutedUsersList, contexts.Server.Id);
```

Comment:
```suggestion
            var mutedUsers = await _usersService.GetUsersAsync(contexts.Server);
            var (title, description, values) = await this.GetMuteEmbedMessage(mutedUsers, contexts.Server.Id);
```

Reply:
ok, a czemu tak jest lepiej?

Reply:
ponieważ
1. robienie listy nie jest wgl potrzebne w tym przypadku
2. tracisz przez to zalety tego że GetUsersAsync jest async

Dzięki temu że GetUsersAsync zwraca IAsyncEnumerable to można rozpocząć obróbkę poszczególnych UserContext zanim wszystkie zostaną pobrane
Czyli:
- pobiera pierwszego usera
- obrabiasz informacje o nim
- w tym samym czasie już pobiera kolejnego
- kiedy pobierze drugiego obrabiasz go
- w tym samym czasie już kolejny
- itd.

Reply:
zamiast:
- czekasz na wszystkich
- obrabiasz wszystkich

Reply:
IAsyncEnumerable jest nowością w c# 8 i pozwala uprościć konstrukcję
Task<IEnumerable>
z tą różnicą, że Task<IEnumerable> też czekał na całą kolekcję, a IAsyncEnumerable pozwala kontynuować od razu po pojawieniu się pierwszego elementu